path/path.go

```diff
@@ -55,19 +87,22 @@ func (s *Resolver) ResolvePath(fpath string) (*merkledag.Node, error) {
 		return nil, err
 	}
 
-	return s.ResolveLinks(nd, parts[1:])
+	return s.ResolveLinks(nd, parts)
 }
 
 // ResolveLinks iteratively resolves names by walking the link hierarchy.
 // Every node is fetched from the DAGService, resolving the next name.
-// Returns the last node found.
+// Returns the list of nodes forming the path, starting with ndd. This list is
+// guaranteed never to be empty.
 //
 // ResolveLinks(nd, []string{"foo", "bar", "baz"})
 // would retrieve "baz" in ("bar" in ("foo" in nd.Links).Links).Links
 func (s *Resolver) ResolveLinks(ndd *merkledag.Node, names []string) (
```

Comment:
@whyrusleeping if you want the signature of `ResolveLinks` not to change, we can move this functionality to a differently-named function.